tests/io/test_bc.py

```diff
@@ -1,43 +1,52 @@
 import inspect
 
 from hydrolib.core.io.bc.models import (
-    Forcing,
-    Function,
-    FunctionData,
+    ForcingBase,
+    ForcingModel,
     TimeInterpolation,
+    TimeSeries,
 )
 from hydrolib.core.io.ini.parser import Parser, ParserConfig
-from ..utils import WrapperTest
+
+from ..utils import WrapperTest, test_data_dir
+
+
+def test_forcing_model():
+    filepath = (
+        test_data_dir
+        / "input/e02/f101_1D-boundaries/c01_steady-state-flow/BoundaryConditions.bc"
+    )
+    m = ForcingModel(filepath)
+    assert len(m.forcing) == 13
+    assert isinstance(m.forcing[-1], TimeSeries)
 
 
 def test_create_a_forcing_from_scratch():
-    forcing = Forcing(
+    forcing = TimeSeries(
         name="F001",
-        function=Function.TimeSeries(
-            time_interpolation=TimeInterpolation.linear,
-            function_data=[
-                FunctionData(quantity="time", unit="s", values=[1.0, 2.0, 3.0])
-            ],
-        ),
+        function="timeseries",
+        timeinterpolation=TimeInterpolation.linear,
+        quantity="time",
+        unit="s",
+        datablock=[[1.0, 2.0, 3.0]],
     )
 
     assert forcing.name == "F001"
-    assert isinstance(forcing.function, Function.TimeSeries)
-    assert forcing.function.offset == 0.0
-    assert forcing.function.factor == 1.0
-    assert forcing.function.time_interpolation == TimeInterpolation.linear
-    assert "time" in forcing.function.function_data
-    assert forcing.function.function_data["time"].quantity == "time"
-    assert forcing.function.function_data["time"].unit == "s"
-    assert forcing.function.function_data["time"].values == [1.0, 2.0, 3.0]
+    assert isinstance(forcing, TimeSeries)
+    assert forcing.offset == 0.0
+    assert forcing.factor == 1.0
+    assert forcing.timeinterpolation == TimeInterpolation.linear
+    assert forcing.quantity[0] == "time"
+    assert forcing.unit[0] == "s"
+    assert forcing.datablock[0] == [1.0, 2.0, 3.0]
 
 
 def test_read_bc_expected_result():
     input_str = inspect.cleandoc(
         """
         [Forcing]
             name              = right01_0001
-            function          = timeSeries
+            function          = timeseries  # what about casing?
```

Comment:
the default we write should be camelCase, but everything is case-insensitive.

Note that this now currently is not a valid bc file, because of the comment.

Reply:
This was a testcase for the lowercase, hence the comment.